src/dashboards/components/AutoRefreshOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+// Libraries
+import React, {FC, useContext, useCallback} from 'react'
+
+// Components
+import {
+  Overlay,
+  Grid,
+  Button,
+  ComponentColor,
+  SelectDropdown,
+} from '@influxdata/clockface'
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+import AutoRefreshDropdown from 'src/shared/components/dropdown_auto_refresh/AutoRefreshDropdown'
+import TimeRangeDropdown from 'src/shared/components/DeleteDataForm/TimeRangeDropdown'
+
+// Types
+import {CustomTimeRange, AutoRefreshStatus} from 'src/types'
+
+// Context
+import AutoRefreshContextProvider, {
+  AutoRefreshContext,
+} from 'src/dashboards/components/RefreshContext'
+
+import './AutoRefresh.scss'
+
+const INACTIVITY_ARRAY = [...Array(25).keys()].map(num => num.toString())
+INACTIVITY_ARRAY[0] = 'None'
+
+export const AutoRefreshForm: FC = () => {
+  const {onClose} = useContext(OverlayContext)
+  const {state, dispatch: setRefreshContext, activateAutoRefresh} = useContext(
+    AutoRefreshContext
+  )
+
+  const handleChooseAutoRefresh = milliseconds => {
+    const status =
+      milliseconds === 0 ? AutoRefreshStatus.Paused : AutoRefreshStatus.Active
+    setRefreshContext({
+      type: 'SET_REFRESH_MILLISECONDS',
+      refreshMilliseconds: {status, interval: milliseconds},
+    })
+  }
+
+  // Because we are not using the manual refresh in the dropdown, this function is a no-op
+  const noop = useCallback((): void => {}, [])
```

Comment:
this is all sorts of ugly. make onManualRefresh an optional param on the AutoRefreshDropdown. you can get rid of the showManualRefresh prop as well. just dont show it if you haven't passed in the optional parameter

Reply:
Great point. This is one of those "can't believe i didnt think of that" moments, missed this bc i was thinking of the ends and not the means here. Thanks for smacking me over the head w this!